docs/user/ppl/admin/catalog.rst

```diff
@@ -0,0 +1,88 @@
+
+
+Definitions of catalog and connector
+====================================
+* Connector is a code component that adapts the query engine to a datasource. For example, Prometheus connector would adapt and help execute the queries to run on Prometheus data source.
+* Catalog is a construct to define how to connect to a datasource and which connector to adapt by query engine.
+
+Example Prometheus Catalog Definition ::
+
+    [{
+        "name" : "prometheus",
+        "connector": "prometheus",
+        "uri" : "http://localhost:9090",
+        "authentication" : {
+            "type" : "basicauth",
+            "username" : "admin",
+            "password" : "admin"
+        }
+    }]
+
+Configuring catalog in OpenSearch.
+====================================
+
+Catalogs are configured in opensearch keystore under `plugins.query.federation.catalog.config` key as they contain credential info.
+A json file containing array of catalog configurations should be injected into keystore with the above mentioned key.
+
+Command to add catalog.json file to OpenSearch Keystore ::
+
+    >>bin/opensearch-keystore add-file plugins.query.federation.catalog.config catalog.json
+
+Catalogs can be configured/overridden during either start up or while the opensearch is running.
+If we update catalog configuration during runtime, the following api should be triggered to update the query engine with the latest changes.
+
+Secure Settings refresh api::
+
+    >> curl --request POST \
+      --url http://{{opensearch-domain}}:9200/_nodes/reload_secure_settings \
+      --header 'content-type: application/json' \
+      --data '{"secure_settings_password":"{{keystore-password}}"}'
+
+
+Catalog configuration Restrictions.
+
+* `name`, `uri`, `connector` are required fields in the catalog configuration.
+* All the catalog names should be unique.
+* Catalog names should match with the regex of an identifier[`[@*A-Za-z]+?[*a-zA-Z_\-0-9]*`].
+* `prometheus` is the only connector allowed.
+
+
+Using a catalog in PPL command.
+====================================
+Catalog is referred in source command in the following way.
+Based on the abstraction designed by the connector,
+one can refer the corresponding entity as table in the source command.
+For example in prometheus connector, each metric is abstracted as a table.
+so we can refer a metric and apply stats over it in the following way.
+
+Example source command with prometheus catalog ::
+
+    >> source = prometheus.prometheus_http_requests_total | stats avg(@value) by job;
+
+
+Limitations of catalog
+====================================
+ * Catalog settings are global and all PPL users are allowed to fetch data from all the defined catalogs.
+ * All data available to the credentials provided in the catalog definition is accessible to PPL users.
+ * With the current release, Basic and AWSSigV4 are the only authentication mechanisms supported with the underlying data sources.
```

Comment:
extra space before `*`?

Reply:
The format seems not correct, content is quoted. https://github.com/vamsi-amazon/sql/blob/docs-prometheus/docs/user/ppl/admin/catalog.rst#id5

Reply:
how does user configure AWSSigV4? add another section in definitions of catalog and connector?

Reply:
There is one more PR coming for authenticators. Will include more details in that PR https://github.com/opensearch-project/sql/pull/929